public/app/plugins/datasource/loki/components/stats.ts

```diff
@@ -52,3 +57,39 @@ export function shouldUpdateStats(
 
   return true;
 }
+
+export function getTimeRange(
+  ds: LokiDatasource,
+  query: LokiQuery,
+  idx: number
+): { start: number | undefined; end: number | undefined } {
+  let start: number, end: number;
+  const NS_IN_MS = 1000000;
+
+  const durationNodes = getNodesFromQuery(query.expr, [Duration]);
+  const durations = durationNodes.map((d) => query.expr.substring(d.from, d.to));
+
+  if (!isLogsQuery(query.expr)) {
+    if (query.queryType === LokiQueryType.Instant) {
+      // metric query with instant type
+      // we want the request timerange to be the query duration e.g. [5m]
+      // with the query -- rate({label="value"} [1h]) -- the range we want to request is: "now - 1h"
+      end = new Date().getTime() * NS_IN_MS;
+      start = end - intervalToMs(durations[idx]) * NS_IN_MS;
```

Comment:
i think the `end` should be the currently-selected-grafana-time-range-end, WDYT?

Reply:
yeah, that makes sense